File: leetcode/66.加一.ts
/*
 * @lc app=leetcode.cn id=66 lang=typescript
 *
 * [66] 加一
 */

// @lc code=start
function plusOne(digits: number[]): number[] {
    let carry = 1;
    for (let i = digits.length - 1; i >= 0; --i) {
        let sum = digits[i] + carry;
        digits[i] = sum % 10;
        carry = Math.floor(sum / 10);
    }
    if (carry > 0) {
        return [carry].concat(digits);
    } else {
        return digits;
    }
};
// @lc code=end

File: leetcode/66.加一.rs
/*
 * @lc app=leetcode.cn id=66 lang=rust
 *
 * [66] 加一
 */

// @lc code=start
impl Solution {
    pub fn plus_one(mut digits: Vec<i32>) -> Vec<i32> {
        let mut carry = 1;

        for digit in digits.iter_mut().rev() {
            *digit += carry;
            carry = *digit / 10;
            *digit %= 10;
        }

        if carry > 0 {
            let mut result = vec![carry];
            result.extend(digits);
            result
        } else {
            digits
        }
    }
}
// @lc code=end
